var/spack/repos/builtin/packages/xsdk/package.py

```diff
@@ -82,7 +82,7 @@ class Xsdk(Package):
     depends_on('sundials@3.1.0~int64+hypre', when='@0.3.0')
     depends_on('sundials@3.1.0~int64+hypre', when='@develop')
 
-    depends_on('plasma@17.1', when='@develop %gcc@4.9:')
+    depends_on('plasma@17.2', when='@develop: %gcc@6.0:')
```

Comment:
i guess you want
```
depends_on('plasma@17.2:', when='@develop %gcc@6.0:')
```
(`17.2:` will be satisfied by `develop` only). Otherwise Spack won't know how to satisfy this constraint.